pkg/api/v1/stats_types.go

```diff
@@ -84,3 +94,56 @@ type PublishRequests struct {
 func (p *PublishRequests) Total() int64 {
 	return p.UnknownPlatform + p.Android + p.IOS
 }
+
+// MarshalCSV returns bytes in CSV format.
+func (s StatsDays) MarshalCSV() ([]byte, error) {
+	// Do nothing if there's no records
+	if len(s) == 0 {
+		return nil, nil
+	}
+
+	var b bytes.Buffer
+	w := csv.NewWriter(&b)
+
+	if err := w.Write([]string{
+		"day",
+		"publish_requests_unknown", "publish_requests_android", "publish_requests_ios",
+		"total_teks_published", "requests_with_revisions", "requests_missing_onset_date", "tek_age_distribution", "onset_to_upload_distribution",
+	}); err != nil {
+		return nil, fmt.Errorf("failed to write CSV header: %w", err)
+	}
+
+	for i, stat := range s {
+		if err := w.Write([]string{
+			stat.Day.Format("2006-01-02"),
+			strconv.FormatInt(stat.PublishRequests.UnknownPlatform, 10),
+			strconv.FormatInt(stat.PublishRequests.Android, 10),
+			strconv.FormatInt(stat.PublishRequests.IOS, 10),
+			strconv.FormatInt(stat.TotalTEKsPublished, 10),
+			strconv.FormatInt(stat.RevisionRequests, 10),
+			strconv.FormatInt(stat.RequestsMissingOnsetDate, 10),
+			join(stat.TEKAgeDistribution, "|"),
+			join(stat.OnsetToUploadDistribution, "|"),
+		}); err != nil {
+			return nil, fmt.Errorf("failed to write CSV entry %d: %w", i, err)
+		}
+	}
+
+	w.Flush()
+	if err := w.Error(); err != nil {
+		return nil, fmt.Errorf("failed to create CSV: %w", err)
+	}
+
+	return b.Bytes(), nil
+}
+
+func join(arr []int64, sep string) string {
```

Comment:
Can be a follow-up, but it'd be good to move this into icsv as a public function in case we need to use it elsewhere.

Reply:
(And probably make "|" the separator as a constant and not a param)

Reply:
@whaught ^